velox/dwio/common/ScanSpec.cpp

```diff
@@ -202,6 +202,50 @@ bool testIntFilter(
   return true;
 }
 
+template <typename T>
+bool testDecimalFilter(
+    common::Filter* filter,
+    dwio::common::DecimalColumnStatistics<T>* decimalStats,
+    bool mayHaveNull) {
+  if (!decimalStats) {
+    return true;
+  }
+
+  if (decimalStats->getMinimum().has_value() &&
```

Comment:
Are we sure the value stored in statistics having the same scale as the filter type?  Could one file store values in one scale and another with a different scale?

Reply:
In parquet file, the schema contains the information of precision and scale, and the statistics have the same scale as the schema type. In orc file, the stats for decimal column are stored as decimal string ([link](https://github.com/apache/orc/blob/d23b42d96176d6e0740118475a32012b5e4e35c7/c%2B%2B/src/Statistics.hh#L713-L723)) so a conversion back to the bigint or hugeint of the schema scale is needed. I'm not clear on how decimal stats are stored in dwrf. Please kindly provide more information or resources if there are.

This PR focuses on parquet and to avoid affecting other file formats, added a `fileFormat` parameter in `testFilter` function to ensure the row group skip only takes effect for parquet. Do you think it makes sense? Thanks.

Reply:
The problem is the scale in `Filter` objects are the table schema type, not necessarily the same as file schema type. So we need to do a check between the two and do rescale if needed.

Reply:
I understand your point. I tested the schema evolution of decimal reader as https://github.com/rui-mo/velox/commit/eae60f5acced0a4c79965d9ae2fbba2809e6ada4 and found the result was incorrect. I suppose we need to support this feature first and I will attempt to find the gap. Thanks.
<Details>
Failed
Expected 20, got 20
20 extra rows, 20 missing rows
10 of extra rows:
	0.10001
	0.10002
	0.10003
	0.10004
	0.10005
	0.10006
	0.10007
	0.10008
	0.10009
	0.10010

10 of missing rows:
	100.01000
	100.02000
	100.03000
	100.04000
	100.05000
	100.06000
	100.07000
	100.08000
	100.09000
	100.10000

Reply:
We cannot check in code that would produce incorrect result.